src/factories.js

```diff
@@ -50,21 +46,19 @@ export function createShorthand(Component, mapValueToProps, val, defaultProps =
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
+    return <Component {...mergePropsAndClassName(mapValueToProps(val), defaultProps)} />
```

Comment:
To clarify my point further, you should be able to rewrite this: 
```js
export function createShorthand(Component, mapValueToProps, val, defaultProps = {}) {
  // Clone ReactElements
  if (isValidElement(val)) {
    return React.cloneElement(val, mergePropsAndClassName(defaultProps, val.props))
  }

  // Create ReactElements from props objects
  if (_.isPlainObject(val)) {
    return <Component {...mergePropsAndClassName(defaultProps, val)} />
  }

  // Map values to props and create a ReactElement
  if (_.isString(val) || _.isNumber(val)) {
    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
  }

  // Otherwise null
  return null
}
```

...as this:
```js
export function createShorthand(Component, mapValueToProps, val, defaultProps = {}) {
  // Clone ReactElements
  if (isValidElement(val)) {
    return React.cloneElement(val, mergePropsAndClassName(defaultProps, val.props))
  }

  // Map values to props if it's a literal
  if (_.isString(val) || _.isNumber(val)) {
    val = mapValueToProps(val)
  }

  // Create ReactElements from props objects
  if (_.isPlainObject(val)) {
    return <Component {...mergePropsAndClassName(defaultProps, val)} />
  }

  // Otherwise null
  return null
}
```

In other words, mapValueToProps is just a convenience that converts a literal to a props object on the end-user's behalf. So it makes no difference whether the user passes their props as an object or as a literal, they should always win.